flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/FileSourceFunction.java

```diff
@@ -123,9 +127,11 @@ public void run(SourceContext<OUT> ctx) throws Exception {
 			OUT nextElement = serializer.createInstance();
 			nextElement =  format.nextRecord(nextElement);
 			if (nextElement == null && splitIterator.hasNext()) {
+				splitCounter.inc();
```

Comment:
the behaviour in both classed is the same. If the format returns null it is considered empty (==completed), hence we increase the counter.

well the FIleSourceFunction doesn't properly call close() once the split was processed.